### PR TITLE
Babel 7 + Latest Flow + Other necessary dependencies

### PR DESCRIPTION
**Pull request by nicarq**
The more time that we take to update to newer versions of Babel and Flow, the more difficult the update is going to be. Currently, we need to improve some things related to Flow.

Upgrades to Babel 7 and Flow
- [x] Upgrades to Babel 7
- [x] Updates/ Changes some node dependencies
- [x] Updates Webpack conf
- [x] Upgrades Flow
- [x] Missing potential real errors from Flow
- [x] Fix Hot Reload

For another PR? (edit: see [here](https://github.com/Emurgo/yoroi-frontend/pull/197))

- [x] Update react-router to 4.x
- [x] Update mobx-react-recouter to latest

If we update react-router, we would be able to fix a annoying issue from the hot reload, which is an error message in the console about not been able to update <Router routes>.

### CODE REVIEW

File: app/App.js [hunk 1]
```diff
@@ -54,3 +55,5 @@ export default class App extends Component<{
     );
   }
 }
+
+export default hot(module)(App);
```
Comment (SebastienGllmt) [inline app/App.js]: are we supposed to do this even in production?

File: package.json [hunk 2]
```diff
@@ -40,19 +40,23 @@
...
+    "@babel/plugin-proposal-class-properties": "7.1.0",
+    "@babel/plugin-proposal-decorators": "7.1.6",
+    "@babel/plugin-proposal-export-default-from": "7.0.0",
+    "@babel/plugin-proposal-export-namespace-from": "7.0.0",
+    "@babel/plugin-transform-runtime": "7.1.0",
+    "@babel/preset-env": "7.1.6",
+    "@babel/preset-flow": "7.0.0",
+    "@babel/preset-react": "7.0.0",
+    "@babel/register": "7.0.0",
+    "@babel/runtime": "7.1.5",
```
[the remark this answers is not in the document]
Comment (SebastienGllmt) [reply package.json]: It looks like even after your latest commit it's there there

[the remark this answers is not in the document]
Comment (nicarq) [reply package.json]: I missed it. I deleted the `package-lock.json` and `@babel/runtime`, and did a clear `npm install` but now it’s throwing lots of errors. It seems that `@babel/runtime-corejs2` doesn’t install all the sub-dependencies that we actually need. I will add it back.

Some of the errors:
```
ERROR in ./app/components/settings/SettingsLayout.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/settings'
 @ ./app/components/settings/SettingsLayout.js 16:57-116
 @ ./app/containers/settings/Settings.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/settings/categories/SupportSettings.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/settings/categories'
 @ ./app/components/settings/categories/SupportSettings.js 18:57-116
 @ ./app/containers/settings/categories/SupportSettingsPage.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/settings/categories/GeneralSettings.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/settings/categories'
 @ ./app/components/settings/categories/GeneralSettings.js 18:57-116
 @ ./app/containers/settings/categories/GeneralSettingsPage.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/settings/categories/TermsOfUseSettings.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/settings/categories'
 @ ./app/components/settings/categories/TermsOfUseSettings.js 16:57-116
 @ ./app/containers/settings/categories/TermsOfUseSettingsPage.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/settings/menu/SettingsMenu.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/settings/menu'
 @ ./app/components/settings/menu/SettingsMenu.js 16:57-116
 @ ./app/containers/settings/Settings.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/settings/menu/SettingsMenuItem.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/settings/menu'
 @ ./app/components/settings/menu/SettingsMenuItem.js 16:57-116
 @ ./app/components/settings/menu/SettingsMenu.js
 @ ./app/containers/settings/Settings.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/sidebar/Sidebar.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/sidebar'
 @ ./app/components/sidebar/Sidebar.js 16:57-116
 @ ./app ^\.\/.*\.js$
 @ ./app/utils/imports.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/sidebar/SidebarCategory.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/sidebar'
 @ ./app/components/sidebar/SidebarCategory.js 16:57-116
 @ ./app ^\.\/.*\.js$
 @ ./app/utils/imports.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/wallet/WalletSendConfirmationDialog.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/wallet'
 @ ./app/components/wallet/WalletSendConfirmationDialog.js 18:57-116
 @ ./app ^\.\/.*\.js$
 @ ./app/utils/imports.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/wallet/WalletBackupDialog.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/wallet'
 @ ./app/components/wallet/WalletBackupDialog.js 16:57-116
 @ ./app/stores/toplevel/WalletBackupStore.js
 @ ./app/stores/index.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/wallet/WalletRestoreDialog.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/wallet'
 @ ./app/components/wallet/WalletRestoreDialog.js 18:57-116
 @ ./app/containers/wallet/WalletAddPage.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/wallet/WalletSendForm.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/wallet'
 @ ./app/components/wallet/WalletSendForm.js 22:57-116
 @ ./app ^\.\/.*\.js$
 @ ./app/utils/imports.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/wallet/WalletAdd.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/wallet'
 @ ./app/components/wallet/WalletAdd.js 18:57-116
 @ ./app/containers/wallet/WalletAddPage.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index

ERROR in ./app/components/wallet/WalletReceive.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/possibleConstructorReturn' in '/Users/narqueros/emurgo/yoroi-icarus/yoroi-frontend/app/components/wallet'
 @ ./app/components/wallet/WalletReceive.js 16:57-116
 @ ./app ^\.\/.*\.js$
 @ ./app/utils/imports.js
 @ ./app/Routes.js
 @ ./app/App.js
 @ ./chrome/extension/index.js
 @ multi ./webpack/customPublicPath webpack-hot-middleware/client?path=__webpack_hmr&dynamicPublicPath=true ./chrome/extension/index
```

[the remark this answers is not in the document]
Comment (nicarq) [reply package.json]: (I just did a rebase to hide those commits)

**Comment by SebastienGllmt** [comment]
Two more things that need to be solved before this gets merged

- [ ] The test `Scenario: Wallet renaming (IT-8)` fails with random chance. The UI updates just fine but for some reason the test hangs.
- [ ] Fix the bug in `trezor-connect` detected by the new version of `flow` (see logs below)
```
node_modules/trezor-connect/lib/popup/PopupManager.js.flow:205:32

string literal `window.close` [1] is incompatible with object type [2].

   node_modules/trezor-connect/lib/popup/PopupManager.js.flow:205:32
   205|         } else if (message === 'window.close') {
                                       ^^^^^^^^^^^^^^ [1]

References:
   node_modules/trezor-connect/lib/popup/PopupManager.js.flow:185:37
   185|     handleExtensionMessage(message: Object): void {
                                            ^^^^^^ [2]
```

**Comment by SebastienGllmt** [comment]
Seems there is a bug with hot-reloading where it crashes if we change anything

**Pull request merged**